### PR TITLE
Improve indexed/searched post types and statuses

### LINKED ISSUE #16: Separate indexable vs. searchable post types
- [ ] Indexing should use `SP_Config::sync_post_types()` to define the indexed post types
- [ ] SP should index all post types so it's not limited to being used just across searchable post types
- [ ] Ensure that the above updates don't affect the post types used when searching

### LINKED ISSUE #31: Index "parent_post_status"
This will simplify post status queries. Props @gibrown

### PR DESCRIPTION
**Pull request by mboynes**
Adds much more flexibility to what gets indexed and searched.

Resolves #16
Resolves #31

### CODE REVIEW

**Comment by mboynes** [comment]
Consider resolving #47 before merging this, since this alters the mapping.

File: lib/class-sp-api.php [hunk 1]
```diff
@@ -158,7 +158,7 @@ public function search( $query, $args = array() ) {
 	}
 
 	public function cluster_health() {
-		$health_uri = apply_filters( 'sp_cluster_health_uri', '/_cluster/health/' . $this->index );
+		$health_uri = apply_filters( 'sp_cluster_health_uri', '/_cluster/health/' . $this->index . '?wait_for_status=yellow&timeout=0.2s' );
```
Comment (joshkadis) [inline lib/class-sp-api.php]: maybe have timeout duration inside a config or `apply_filters()`?

File: lib/class-sp-search.php [hunk 2]
```diff
@@ -114,8 +114,8 @@ public function get_posts() {
 		} else {
 			$ids = $this->pluck_field( 'post_id' );
 			$this->posts = get_posts( array(
-				'post_type'      => 'any',
-				'post_status'    => 'any',
+				'post_type'      => array_values( get_post_types() ),
```
Comment (timatron) [inline lib/class-sp-search.php]: i would flip these two back to 'any', if we're doing a query by $id, there should be nothing else to impede that in some edge case where there is an id we want that for some reason had a post type or status that isn't returned by those functions

File: lib/class-sp-api.php (hunk 1 above)
Comment (mboynes) [inline lib/class-sp-api.php]: I thought about doing that, but it ultimately is being filtered. If someone wanted to change it, they could do so through this filter (`sp_cluster_health_uri`). I'll add a docblock highlighting that.

**Pull request merged**